platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheGroovyClosureIntegrationTest.groovy

```diff
@@ -158,11 +201,12 @@ class ConfigurationCacheGroovyClosureIntegrationTest extends AbstractConfigurati
                     doFirst {
                         println(name) // task property is ok
                         println(gradleVersion)
+                        throw new IllegalStateException("UNREACHABLE")
                     }
                 }
             }
         """
-        executer.beforeExecute { withArguments("-I", initScript.absolutePath) }
+        executer.beforeExecute { withArgument("-I").withArgument(initScript.absolutePath) }
```

Comment:
💭 Perhaps counter-intuitively, the `withArguments` method resets the arguments. As one effect, it throws away the args required to capture build operations, needed for CC fixtures. At the same time, the `withArgument` method _adds_ exactly one argument...

I think it's not the first time I stumble into this. We should polish this at some point.